lrlex/src/lib/ctbuilder.rs

```diff
@@ -226,7 +226,7 @@ where
     LexerTypesT::StorageT: Debug + Eq + Hash + ToTokens,
     usize: num_traits::AsPrimitive<LexerTypesT::StorageT>,
 {
-    lrpar_config: Option<Box<dyn Fn(CTParserBuilder<LexerTypesT>) -> CTParserBuilder<LexerTypesT>>>,
+    lrpar_config: Option<Box<dyn Fn(CTParserBuilder<LexerTypesT>) -> CTParserBuilder<LexerTypesT> + 'a>>,
```

Comment:
If we're willing to change the lifetime of this closure from being `'static` to another lifetime like `'a` here,
we can entirely get rid of the leak variables hacks.

This is here because Box<dyn Trait> is defaulting to 'static unless otherwise specified.
With it specified to something like 'a, the closure no longer needs to outlive the builder and main.

With that we can get rid of the leak hacks in this branch (at least, haven't looked into further cleanups yet)

Reply:
Going from `'static` -> `'a` seems fully backwards compatible (i.e. it's more general) to me, so I don't see a problem with this!